### PR TITLE
pythonPackages.patator: init at 0.7

### PR DESCRIPTION
**Pull request by y0no**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/patator/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	ajpy
+	pyopenssl
+	cx_Oracle
+	mysqlclient
+	psycopg2
+	pycrypto
+	dnspython
+	IPy
+	pysnmp
+	pyasn1
```
Comment (dotlambda) [inline pkgs/development/python-modules/patator/default.nix]: indentation

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -9183,24 +9187,9 @@ in {
...
-    doCheck = false;
-
-    meta = {
-      description = "Class and tools for handling of IPv4 and IPv6 addresses and networks";
-      homepage = https://pypi.python.org/pypi/IPy;
-    };
-  };
+  IPy = callPackage ../development/python-modules/IPy { };
+  # Alias to remove 
+  ipy = callPackage ../development/python-modules/IPy { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: What's wrong with the old name? Most attributes in Nixpkgs are lowercase.

File: pkgs/development/python-modules/cx_oracle/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    oracle-instantclient
+  ];
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "80545fc7acbdda917dd2b1604c938141256bdfed3ad464a44586c9c2f09c3004";
+  };
+  
+  checkPhase = ''
+    LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/oracle/12.2/client64/lib
```
Comment (dotlambda) [inline pkgs/development/python-modules/cx_oracle/default.nix]: :question:

File: pkgs/development/python-modules/patator/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  version = "0.7";
+  name = "${pname}-${version}";
+  disabled = !(isPy3k);
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "335e432e6cc591437e316ba8c1da935484ca39fc79e595ccf60ccd9166e965f1";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
```
Comment (dotlambda) [inline pkgs/development/python-modules/patator/default.nix]: Please add the dependencies as arguments instead of `python3Packages`.

**Review by dotlambda — CHANGES_REQUESTED** [review]
`name` is not necessary for `buildPython*` anymore. Please remove it

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (y0no) [reply pkgs/top-level/python-packages.nix]: Don't know what's better between the real name (as a lots of packages), or all in lowercase. I think, in pypi you can have two different packages with the same name except but different capitalized letters.

File: pkgs/development/python-modules/patator/default.nix (hunk 4 above)
Comment (y0no) [reply pkgs/development/python-modules/patator/default.nix]: Why not. What's the difference ?

File: pkgs/development/python-modules/patator/default.nix (hunk 1 above)
Comment (y0no) [reply pkgs/development/python-modules/patator/default.nix]: Thanks, I fix this in next commit =)

File: pkgs/development/python-modules/IPy/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+  pname = "IPy";
+  version = "0.83";
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/IPy/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/ajpy/default.nix [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+  pname = "ajpy";
+  version = "0.0.2";
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/ajpy/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/cx_oracle/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, buildPythonPackage, fetchPypi, oracle-instantclient }:
+
+buildPythonPackage rec {
+  pname = "cx_Oracle";
+  version = "6.1";
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/cx_oracle/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/cx_oracle/default.nix (hunk 3 above)
Comment (FRidh) [reply pkgs/development/python-modules/cx_oracle/default.nix]: You are referring to a path outside the store. That is incorrect. Furthermore, paths to libraries are better hardcoded. Also, something should *run* in the `checkPhase`.

File: pkgs/development/python-modules/mysqlclient/default.nix [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, buildPythonPackage, fetchPypi, libmysql }:
+
+buildPythonPackage rec {
+  pname = "mysqlclient";
+  version = "1.3.12";
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/mysqlclient/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/patator/default.nix [hunk 9]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, buildPythonPackage, isPy3k, python3Packages, fetchPypi }:
+
+buildPythonPackage rec {
+  pname = "patator";
+  version = "0.7";
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/patator/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/patator/default.nix [hunk 10]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, buildPythonPackage, isPy3k, python3Packages, fetchPypi }:
```
Comment (FRidh) [inline pkgs/development/python-modules/patator/default.nix]: Don't pass in the package set; instead pass in individual requirements.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (FRidh) [reply pkgs/top-level/python-packages.nix]: No, that's not true. The attributes are preferably the normalized names, so `ipy` in this case.
https://www.python.org/dev/peps/pep-0503/#normalized-names

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/cx_oracle/default.nix (hunk 3 above)
Comment (y0no) [reply pkgs/development/python-modules/cx_oracle/default.nix]: Okay, Is there a good way to handle the path?

**Comment by y0no** [comment]
Hope the changes requested are well fixed.

**Comment by dotlambda** [comment]
You're still not running any checks in the `checkPhase`.
Also, please squash the commits.

**Comment by y0no** [comment]
Woops you are right. In fact, it seems that cx_Oracle tests need an Oracle database running in background. I think it will be better to pass the test if we don't want an error

**Comment by dotlambda** [comment]
Yes, if it needs a database, then you can set `doCheck = false`. But don't forget to add a comment with the reason.

**Comment by dotlambda** [comment]
Now, it would be nice if you replaced your 11 commits by 5, one for each package.
And please name your commits according to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md), e.g. `pythonPackages.mysqlclient: init at 1.3.12`.

**Comment by y0no** [comment]
Ok, I think it's good now.

**Comment by dotlambda** [comment]
Very nice. Thank you! I'll try to build them...

@GrahamcOfBorg build python2.pkgs.ipy python3.pkgs.ipy
@GrahamcOfBorg build python2.pkgs.ajpy python3.pkgs.ajpy
@GrahamcOfBorg build python2.pkgs.cx_oracle python3.pkgs.cx_oracle
@GrahamcOfBorg build python2.pkgs.mysqlclient python3.pkgs.mysqlclient
@GrahamcOfBorg build python3.pkgs.patator

**Comment by dotlambda** [comment]
> Ran 0 tests in 0.000s

Do ajpy and mysqlclient provide tests? If so, you should specify the appropriate `checkPhase`. Otherwise, please set `doCheck = false` and add a comment that there are no tests.

Regarding ipy, the tests don't seem to run there as well. Do you have an idea why?
Maybe use `nosetests` in the `checkPhase`? See https://github.com/autocracy/python-ipy/blob/master/.travis.yml
Another idea would be to use `make -B`.

**Comment by y0no** [comment]
Ajpy, patator doesn't seems to got tests.
Concerning mysqlclient, it's like cx_Oracle, we need a database to run the tests.
Finally, I have fixed the tests for IPy. As you mentionned, the tests are run with nose.

**Comment by FRidh** [comment]
A dependency can't be built:
```sh
nix build python3.pkgs.cx_oracle -f .
builder for '/nix/store/6k80zw21b9chd00lqya0vfj03y8s91xw-oracle-instantclient12.2-basic-12.2.0.1.0-1.x86_64.rpm.drv' failed to produce output path '/nix/store/3ly42sshg74lqk1vw2fksagq6zd3sz0g-oracle-instantclient12.2-basic-12.2.0.1.0-1.x86_64.rpm'
cannot build derivation '/nix/store/1g1bbai4y39zx3b4y26c2waip8ymygdf-oracle-instantclient-12.2.0.1.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/pqy6859y8xs0gkd5758faz28aff6300d-python3.6-cx_Oracle-6.1.drv': 1 dependencies couldn't be built
[0 built (1 failed)]
error: build of '/nix/store/pqy6859y8xs0gkd5758faz28aff6300d-python3.6-cx_Oracle-6.1.drv' failed
```

**Comment by y0no** [comment]
Can this error be made because of oracle-instantclient license ? It correctly build on my laptop when oracle-instantclient is installed (package that I don't handle)

**Comment by FRidh** [comment]
Right, it needs prefetching. That important part was not visible in the `nix build` output. You can ignore it.

**Comment by y0no** [comment]
So do I have to do something else to get the package merged ?

**Comment by dotlambda** [comment]
@FRidh This looks fine, doesn't it? I did however not test if it works.

**Comment by dotlambda** [comment]
ping @FRidh

**Comment by dotlambda** [comment]
@y0no Can you rebase on master?
@FRidh Any objections against merging?

**Review by FRidh — APPROVED** [review]

**Review by dotlambda — APPROVED** [review]

**Pull request merged**